### PR TITLE
Update release process notes

### PR DESCRIPTION
**Pull request by hsaraogi**
---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/825)

### CODE REVIEW

File: docs/source/release_notes/release-process.rst [hunk 1]
```diff
@@ -24,12 +24,8 @@ Cutting for Release
 2. Tag the commit: ``git tag -a <version>``.
 3. Push the tag to Github:
    ``git push origin <version>:refs/tags/<version>``.
-4. After a short wait you can publish the artifacts on Bintray by going
+4. After a short wait (when the build at circleCI is successful) the artifacts will be published on BinTray
    to the `AtlasDB
    page <https://bintray.com/palantir/releases/atlasdb/view>`__.
-
-   .. note::
-      You must have admin privledges on Bintray for this to work.
```
Comment (rhero) [inline docs/source/release_notes/release-process.rst]: Do you need to have admin privileges to tag or push this up? This reads like anybody can create official releases of AtlasDB (and I don't know if that's intentional).

Comment (hsaraogi) [inline docs/source/release_notes/release-process.rst]: Anyone with write permissions can Create and edit releases. The `Palantir Users`(241 members)  group has write permissions and will be able to release.

Comment (tpetracca) [inline docs/source/release_notes/release-process.rst]: we should try to protect releases/tags the same way we protect the develop branch (i.e. only the atlas team can write to it)

Comment (rhero) [inline docs/source/release_notes/release-process.rst]: @SerialVelocity says it doesn't look like it's possible to separate write permissions from tagging, or prevent people from pushing to <version>:refs/tags/<version>. So I guess Palantir Users can do it.

**Pull request merged**